src/app/page.tsx

```diff
@@ -1,41 +1,37 @@
-import {
-	EstimatedPackage,
-	PackageOwnership,
-	tideliftMeUp,
-} from "tidelift-me-up";
-
 import { Footer } from "~/components/Footer";
 import { MainArea } from "~/components/MainArea";
 import { OptionsForm } from "~/components/OptionsForm";
 import { ResultDisplay } from "~/components/ResultDisplay";
 import { ScrollButton } from "~/components/ScrollButton";
+import { SearchParamsType, fetchData } from "~/utils/fetchData";
 
+import { metadata as defaultMetadata } from "./layout";
 import styles from "./page.module.css";
 
 export interface HomeProps {
-	searchParams: Record<string, unknown>;
+	searchParams: SearchParamsType;
 }
 
-export default async function Home({ searchParams }: HomeProps) {
-	const options = {
-		ownership: undefinedIfEmpty(
-			[
-				searchParams["author"] === "on" && "author",
-				searchParams["maintainer"] === "on" && "maintainer",
-				searchParams["publisher"] === "on" && "publisher",
-			].filter(Boolean) as PackageOwnership[],
-		),
-		since: (searchParams["since"] || undefined) as string | undefined,
-		username: searchParams.username as string,
-	};
-	let result: Error | EstimatedPackage[] | undefined;
+export async function generateMetadata({ searchParams }: HomeProps) {
+	const { options, result } = await fetchData(searchParams);
+	const username = options.username || "";
+	const packageCount = Array.isArray(result) ? result.length : 0;
```

Comment:
[Bug] If `fetchData` returns an `Error`, we'd want to give a specific error-y description. Like a _"could not find packages for ..."_ or something like that.

Btw, just so I don't forget them later, I think these are the five total cases that need to be handled:

* Unknown error from the try/catch in `fetchData`
* Username does not exist (blocked, see other comment on this line)
* 0 packages found
* 1 package found
* Multiple packages found